File: soap/Q2242_DataMinMaxGcdSpan.java
import java.util.Arrays;
import java.util.List;
import soap.q2211.generated.DataService;
import soap.q2211.generated.SoapDataService;

public class Q2242_DataMinMaxGcdSpan {
    static final String STUDENT_CODE = "B21DCCN001";   // TODO
    static final String Q_CODE       = "TODO_qCode";   // TODO

    public static void main(String[] args) {
        DataService service = new DataService();
        SoapDataService port = service.getSoapDataServicePort();

        List<Integer> data = port.getData(STUDENT_CODE, Q_CODE);
        System.out.println("data=" + data);

        int min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
        int gcd = 0;
        for (int v : data) {
            if (v < min) min = v;
            if (v > max) max = v;
            gcd = gcd(gcd, Math.abs(v));
        }
        int span = max - min;
        List<Integer> answer = Arrays.asList(min, max, gcd, span);
        System.out.println("answer=" + answer);

        port.submitDataIntArray(STUDENT_CODE, Q_CODE, answer);
        System.out.println("Đã submit " + answer);
    }

    static int gcd(int a, int b) {
        while (b != 0) { int t = b; b = a % b; a = t; }
        return a;
    }
}

File: soap/Q2242_DataMinMaxGcdSpan.py
from math import gcd
from functools import reduce
from zeep import Client

EXAM_IP      = "36.50.135.242"
STUDENT_CODE = "B22DCCN863"   # TODO
Q_CODE       = "TODO_qCode"   # TODO
WSDL         = f"http://{EXAM_IP}:2221/DataService?wsdl"


def main():
    client = Client(WSDL)

    data = [int(v) for v in client.service.getData(STUDENT_CODE, Q_CODE)]
    print("data=", data)

    mn   = min(data)
    mx   = max(data)
    g    = reduce(gcd, (abs(v) for v in data), 0)
    span = mx - mn
    answer = [mn, mx, g, span]
    print("answer=", answer)

    client.service.submitDataIntArray(STUDENT_CODE, Q_CODE, answer)
    print("Đã submit", answer)


if __name__ == "__main__":
    main()
